gen/enum.go

```diff
@@ -265,6 +275,34 @@ func enumItemName(enumName string, spec *compile.EnumItem) (string, error) {
 	return enumName + name, err
 }
 
+// enumItemLabelName returns the actual name used for serialization/deserialization
+// default to EnumItem.Name, override by the value of GoLabel
+func enumItemLabelName(spec *compile.EnumItem) string {
+	labelName := spec.Name
+	val, ok := spec.Annotations[GoLabel]
+	if ok && len(val) > 0 {
+		labelName = val
+	}
+	return labelName
+}
+
+// validateEnumUniqueNames apply name label GoLabel and raise error if there's
+// duplicates in resolved enum item names
+func validateEnumUniqueNames(spec *compile.EnumSpec) error {
+	items := spec.Items
+	used := make(map[string]bool, len(items))
+	for _, i := range items {
+		itemName := enumItemLabelName(&i)
+		if _, isUsed := used[itemName]; isUsed {
+			return fmt.Errorf(
+				"duplicated item name %q found in enum %q",
```

Comment:
nit: If we find a conflict, it would be nice to know what it's with. The above
map could be changed to a `map[string]string`, mapping label to original item.

    map[string]EnumItem

Then this can be,

    if conflict, isUsed := used[itemName]; isUsed {
        return fmt.Errorf(
            "item %q with label %q conflicts with item %q in enum %q",
            i.Name, itemName, conflict.Name, spec.Name)
    }

Reply:
makes sense